pulsar-functions/instance/src/main/java/org/apache/pulsar/functions/instance/InstanceUtils.java

```diff
@@ -89,23 +85,26 @@
         }
     }
 
-    public ComponentType calculateSubjectType(Function.FunctionDetails functionDetails) {
+    public Function.FunctionDetails.ComponentType calculateSubjectType(Function.FunctionDetails functionDetails) {
```

Comment:
can we add a unit test for this to make sure there isn't any issues with bc

Reply:
unittest added